app/connectors/FinancialStatementConnector.scala

```diff
@@ -38,7 +40,24 @@ class FinancialStatementConnector @Inject()(http: HttpClient, config: FrontendAp
     http.GET[HttpResponse](url)(implicitly, schemeHc, implicitly).map { response =>
       response.status match {
         case OK =>
-          response.json.as[Seq[PsaFS]]
+          response.json.as[Seq[PsaFS]].filterNot(_.chargeType == PsaFSChargeType.PAYMENT_ON_ACCOUNT)
+        case _ =>
+          handleErrorResponse("GET", url)(response)
+      }
+    }
+  }
+
+  def getPsaFSWithoutAndWithPaymentOnAccount(psaId: String)
```

Comment:
I'm not sure I like this way of doing this. Why not just filter in the controller since it is only used in one place? It just seems pointless filtering in here when it could be done where it is used